tests/functional/test_regions.py

```diff
@@ -546,3 +546,17 @@ def test_unregionalized_client_with_unknown_region(self):
         self.assertTrue(
             stubber.requests[0].url.startswith('https://iam.amazonaws.com/')
         )
+
+
+@pytest.mark.parametrize("is_builtin", [True, False])
+def test_endpoint_resolver_knows_its_datasource(is_builtin):
+    # The information whether or not the endpoints.json file was loaded from
+    # the builtin data directory or not should be passed from Loader to
+    # EndpointResolver.
+    session = _get_patched_session()
+    loader = session.get_component('data_loader')
+    with mock.patch.object(
+        loader, 'is_builtin_path', mock.Mock(return_value=is_builtin)
+    ):
```

Comment:
Turning https://github.com/boto/botocore/pull/2751/files#r966037688 into suggestion for easy batching in Github UI:

```suggestion
    with mock.patch.object(loader, 'is_builtin_path', return_value=is_builtin):
```